lib/srv/desktop/rdp/rdpclient/client.go

```diff
@@ -385,284 +388,310 @@ func (c *Client) startInputStreaming(stopCh chan struct{}) error {
 		}
 
 		if atomic.LoadUint32(&c.readyForInput) == 0 {
-			// Input not allowed yet, drop the message.
-			c.cfg.Logger.DebugContext(context.Background(), "Dropping TDP input message, not ready for input")
+			switch m := msg.(type) {
+			case tdp.ClientScreenSpec:
+				// Withhold the latest screen size until the client is ready for input. This ensures
+				// that the client receives the correct screen size when it is ready.
+				withheldResize = &m
+				c.cfg.Logger.DebugContext(context.Background(), "Withholding screen size until client is ready for input", "width", m.Width, "height", m.Height)
+			default:
+				// Ignore all messages except ClientScreenSpec until the client is ready for input.
+				c.cfg.Logger.DebugContext(context.Background(), "Dropping TDP input message, not ready for input")
+			}
+
 			continue
 		}
 
 		c.UpdateClientActivity()
 
-		switch m := msg.(type) {
-		case tdp.ClientScreenSpec:
-			// If the client has specified a fixed screen size, we don't
-			// need to send a screen resize event.
-			if c.cfg.hasSizeOverride() {
-				continue
+		if withheldResize != nil {
+			c.cfg.Logger.DebugContext(context.Background(), "Sending withheld screen size to client")
+			if err := c.handleMessage(*withheldResize); err != nil {
+				return trace.Wrap(err)
 			}
+			withheldResize = nil
+		}
 
-			c.cfg.Logger.DebugContext(context.Background(), "Client changed screen size", "width", m.Width, "height", m.Height)
-			if errCode := C.client_write_screen_resize(
-				C.ulong(c.handle),
-				C.uint32_t(m.Width),
-				C.uint32_t(m.Height),
-			); errCode != C.ErrCodeSuccess {
-				return trace.Errorf("ClientScreenSpec: client_write_screen_resize: %v", errCode)
-			}
-		case tdp.MouseMove:
-			mouseX, mouseY = m.X, m.Y
-			if errCode := C.client_write_rdp_pointer(
-				C.ulong(c.handle),
-				C.CGOMousePointerEvent{
-					x:      C.uint16_t(m.X),
-					y:      C.uint16_t(m.Y),
-					button: C.PointerButtonNone,
-					wheel:  C.PointerWheelNone,
-				},
-			); errCode != C.ErrCodeSuccess {
-				return trace.Errorf("MouseMove: client_write_rdp_pointer: %v", errCode)
-			}
-		case tdp.MouseButton:
-			// Map the button to a C enum value.
-			var button C.CGOPointerButton
-			switch m.Button {
-			case tdp.LeftMouseButton:
-				button = C.PointerButtonLeft
-			case tdp.RightMouseButton:
-				button = C.PointerButtonRight
-			case tdp.MiddleMouseButton:
-				button = C.PointerButtonMiddle
-			default:
-				button = C.PointerButtonNone
-			}
-			if errCode := C.client_write_rdp_pointer(
+		if err := c.handleMessage(msg); err != nil {
+			return trace.Wrap(err)
+		}
+	}
+}
+
+func (c *Client) handleMessage(msg tdp.Message) error {
```

Comment:
`handleMessage` seems too vague of a name - is it an incoming message or an outgoing message?

Reply:
f36db5e5f1c7dcaa346b1438557a8b6005c5ba35